clientconn.go

```diff
@@ -414,6 +432,9 @@ func (cc *ClientConn) newAddrConn(addr Address, skipWait bool) error {
 		go func() {
 			if err := ac.resetTransport(false); err != nil {
 				grpclog.Printf("Failed to dial %s: %v; please retry.", ac.addr.Addr, err)
+				ac.cc.mu.Lock()
```

Comment:
seems like all of this can be inside of `ac.resetTransport` when called with `false`.